flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/SourceFunction.java

```diff
@@ -34,17 +34,17 @@
  * The run method can run for as long as necessary. The source must, however, react to an
  * invocation of {@link #cancel()} by breaking out of its main loop.
  *
- * <h3>Checkpointed Sources</h3>
+ * <h3>CheckpointedFunction Sources</h3>
  *
- * <p>Sources that also implement the {@link org.apache.flink.streaming.api.checkpoint.Checkpointed}
+ * <p>Sources that also implement the {@link org.apache.flink.streaming.api.checkpoint.CheckpointedFunction}
  * interface must ensure that state checkpointing, updating of internal state and emission of
  * elements are not done concurrently. This is achieved by using the provided checkpointing lock
  * object to protect update of state and emission of elements in a synchronized block.
  *
  * <p>This is the basic pattern one should follow when implementing a (checkpointed) source:
  *
  * <pre>{@code
- *  public class ExampleSource<T> implements SourceFunction<T>, Checkpointed<Long> {
+ *  public class ExampleSource<T> implements SourceFunction<T>, CheckpointedFunction<Long> {
```

Comment:
Could you please also update the methods of the class in the code examples? `snapshotState` and `restoreState` have changed with the interface change.

Reply:
Please check, I have made the changes, but could not make up what to write inside of these functions

Reply:
You can see the function bodies here:
https://ci.apache.org/projects/flink/flink-docs-release-1.4/dev/stream/state/state.html#stateful-source-functions